pkg/api/v1/defaults.go

```diff
@@ -101,6 +101,14 @@ func SetDefaults_Service(obj *v1.Service) {
 	if obj.Spec.SessionAffinity == "" {
 		obj.Spec.SessionAffinity = v1.ServiceAffinityNone
 	}
+	if obj.Spec.SessionAffinity == v1.ServiceAffinityClientIP && obj.Spec.SessionAffinityConfig == nil {
```

Comment:
Don't forget `|| obj.Spec.SessionAffinityConfig.ClientIP == nil || obj.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == 0`

Reply:
Yes, I agree. But, do you mean `obj.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == nil`?

`TimeoutSeconds` is `*int32` type and we treat `TimeoutSeconds == 0` as an invalid value, ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L1913